mat64/symmetric.go

```diff
@@ -51,25 +51,29 @@ type MutableSymmetric interface {
 	SetSym(i, j int, v float64)
 }
 
-// NewSymDense constructs an n x n symmetric matrix. If len(mat) == n * n,
-// mat will be used to hold the underlying data, or if mat == nil, new data will be allocated.
-// The underlying data representation is the same as a Dense matrix, except
-// the values of the entries in the lower triangular portion are completely ignored.
-func NewSymDense(n int, mat []float64) *SymDense {
+// NewSymDense creates a new Symmetric matrix with n rows and columns. If
+// len(data) == n*n, data is used as the underlying slice (changes to the matrix
```

Comment:
"data is used as the backing slice. ... Changes to elements of the returned SymDense will be reflected in data."